### ISSUE TITLE
Automatically generated mock objects passed as constructor parameters of a mock don't get stubbed unit returning functions

### ISSUE THREAD

**Issue by riQQ** [mockative/mockative#122]
Automatically mocked objects passed as constructor parameters of a mock don't get stubbed unit returning functions even though `stubsUnitByDefault` is `true`. Is this on purpose?

Given a mockable type *A* with a property of the mockable type *B*, `stubsUnitByDefault` is `true` for an object of type *A*, but  `false` for the object of type *B* passed to the constructor of type *A*.
The generated `mock()` function configures the mock itself (type *A*) setting `stubsUnitByDefault`, but the generated code for mock class *A* doesn't set `stubsUnitByDefault` for the mock object of type *B* passed to the constructor of type *A*. A `mock()` function for type *B* is not generated.

Below is a failing test for this scenario. I can open a PR with this code, if this is something that is supposed to work.

```kotlin
package io.github

import io.mockative.Mock
import io.mockative.classOf
import io.mockative.mock
import io.mockative.verify
import kotlin.test.Test

@Mockable
class MockWithNestedMockA(val nestedMock: NestedClass) {
    init {
        nestedMock.doSomethingReturnUnit()
    }
}

@Mockable
class NestedClass {
    fun doSomethingReturnUnit() {

    }
}

class MockWithNestedMockWithUnitReturningFunctionTests {
    @Mock
    val mockedObjectWithNestedMock = mock(classOf<MockWithNestedMockA>())

    @Test
    fun does_not_match_if_one_doesnt_match() {
        verify { mockedObjectWithNestedMock.nestedMock.doSomethingReturnUnit() }.wasInvoked(1)
    }
}

```

**Comment by kris098e** [comment]
Hi @riQQ, thanks for your contribution once again :)!

I haven't given it a thought, but if the argument to KSP for stubbing units by default is true, then it should also be the case for transitive mocks. 

Stubbing units by default otherwise should, in my opinion, then be passed as an argument to the annotated mock, i.e something like

```kotlin
@Mock
val mockedClassWithProperty = mock(classOf<ClassWithProperty>(), transitiveStubUnitByDefault = true)
```

**Comment by riQQ** [comment]
I agree. I've opened a PR with the failing test.

**Comment by riQQ** [comment]
No, as the issue is not with the mock itself but the mock's property that is mocked recursively.
Running the test from #123 with your workaround still fails with the same error.

**Comment by kris098e** [comment]
Hi @riQQ 

Looks good! I have added some comments
